src/ORM/Association/HasMany.php

```diff
@@ -406,23 +411,27 @@ function ($v) {
      * @param array $foreignKey array of foreign key properties
      * @param Table $target The associated table
      * @param array $conditions The conditions that specifies what are the objects to be unlinked
-     * @return void
+     * @return bool success
      */
     protected function _unlink(array $foreignKey, Table $target, array $conditions = [])
     {
         $mustBeDependent = (!$this->_foreignKeyAcceptsNull($target, $foreignKey) || $this->dependent());
+
         if ($mustBeDependent) {
             if ($this->_cascadeCallbacks) {
                 $query = $this->find('all')->where($conditions);
+                $ok = true;
                 foreach ($query as $assoc) {
-                    $target->delete($assoc);
+                    $ok = $ok && $target->delete($assoc);
                 }
+                return $ok;
             } else {
                 $target->deleteAll($conditions);
+                return true;
             }
         } else {
             $updateFields = array_fill_keys($foreignKey, null);
-            $target->updateAll($updateFields, $conditions);
+            return $target->updateAll($updateFields, $conditions);
 
```

Comment:
Does updateAll return rows effected or the success? For some reason I thought it returned rows effected.